packages/plugin-client-drizzle/src/pg/session.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  Assume,
+  Logger,
+  NoopLogger,
+  Query,
+  RelationalSchemaConfig,
+  SelectedFieldsOrdered,
+  TablesRelationalConfig,
+  entityKind,
+  fillPlaceholders,
+  sql
+} from 'drizzle-orm';
+import {
+  PgColumn,
+  PgDialect,
+  PgSession,
+  PgTransaction,
+  PgTransactionConfig,
+  PgPreparedQuery,
+  PreparedQueryConfig,
+  QueryResultHKT
+} from 'drizzle-orm/pg-core';
+import { Client, Pool, PoolClient, QueryArrayConfig, QueryConfig, QueryResult, QueryResultRow } from 'pg';
+import { mapResultRow } from '../shared/utils';
+
+export type XataClient = Pool | PoolClient | Client;
+
+export class XataPreparedQuery<T extends PreparedQueryConfig> extends PgPreparedQuery<T> {
+  static readonly [entityKind]: string = 'XataPreparedQuery';
+
+  private rawQueryConfig: QueryConfig;
+  private queryConfig: QueryArrayConfig;
+
+  constructor(
+    private client: XataClient,
+    queryString: string,
+    private params: unknown[],
+    private logger: Logger,
+    private fields: SelectedFieldsOrdered<PgColumn> | undefined,
+    name: string | undefined,
+    private customResultMapper?: (rows: unknown[][]) => T['execute']
+  ) {
+    super({ sql: queryString, params });
+    this.rawQueryConfig = {
+      name,
+      text: queryString
+    };
+    this.queryConfig = {
+      name,
+      text: queryString,
+      rowMode: 'array'
+    };
+  }
+
+  async execute(placeholderValues: Record<string, unknown> | undefined = {}): Promise<T['execute']> {
+    const params = fillPlaceholders(this.params, placeholderValues);
+
+    this.logger.logQuery(this.rawQueryConfig.text, params);
+
+    const {
+      fields,
+      client,
+      rawQueryConfig: rawQuery,
+      queryConfig: query,
+      // @ts-expect-error joinsNotNullableMap is internal
+      joinsNotNullableMap,
+      customResultMapper
+    } = this;
+    if (!fields && !customResultMapper) {
+      return await client.query(rawQuery, params);
+    }
+
+    const result = await client.query(query, params);
+
+    return customResultMapper
```

Comment:
Can your remove the `!` from fields since you already checked for it's existence on L69?